### PR TITLE
Ensure bond is sufficient to potentially refund openKeep fee 

### PR DESCRIPTION
**Pull request by NicholasDotSol**
Revert on deposit creation if the keep does not have sufficient bond to refund open-keep fee in case of signer setup failure.

### CODE REVIEW

File: solidity/contracts/deposit/DepositUtils.sol [hunk 1]
```diff
@@ -438,9 +438,12 @@ library DepositUtils {
...
         require(address(this).balance >= _ethValue, "Not enough funds to send");
-        IBondedECDSAKeep _keep = IBondedECDSAKeep(_d.keepAddress);
-        _keep.returnPartialSignerBonds.value(_ethValue)();
-        return true;
+        if(_ethValue > 0){
+            IBondedECDSAKeep _keep = IBondedECDSAKeep(_d.keepAddress);
+            _keep.returnPartialSignerBonds.value(_ethValue)();
+            return true;
+        }
+        return false;
```
Comment (Shadowfiend) [inline solidity/contracts/deposit/DepositUtils.sol:446]: This is going to fall under the return thing we're doing in the other PR, but slightly different: this function only used to ever return true… But also no one pays attention to the return value. I think we can drop the return value.

File: solidity/test/DepositFundingTest.js [hunk 2]
```diff
@@ -125,6 +130,27 @@ describe("DepositFunding", async function() {
       )
     })
 
+    it("reverts if bond is insufficient to cover a deposit creation fee refund", async () => {
+      const depositFee = await tbtcSystemStub.getNewDepositFeeEstimate()
+
+      await ecdsaKeepStub.send(depositFee - 1)
+      await ecdsaKeepStub.setBondAmount(depositFee - 1)
```
Comment (Shadowfiend) [inline solidity/test/DepositFundingTest.js:137]: Head-whirly that we don't have to mess with the ETHBTC price here, but more illustrative of how complicated all the bits are.

Comment (NicholasDotSol) [reply solidity/test/DepositFundingTest.js:137]: yeah it's definitely gotten incrementally more complicate

File: solidity/contracts/deposit/DepositFunding.sol [hunk 3]
```diff
@@ -93,9 +96,11 @@ library DepositFunding {
         // refund the deposit owner the cost to create a new Deposit at the time the Deposit was opened.
         uint256 _seized = _d.seizeSignerBonds();
 
-        /* solium-disable-next-line security/no-send */
-        _d.enableWithdrawal(_d.depositOwner(), _d.keepSetupFee);
-        _d.pushFundsToKeepGroup(_seized.sub(_d.keepSetupFee));
+        if(_seized >= _d.keepSetupFee){
+            /* solium-disable-next-line security/no-send */
```
Comment (Shadowfiend) [inline solidity/contracts/deposit/DepositFunding.sol:100]: Well this obviously no longer applies :p

**Pull request merged**